test/integrationtests/lab_interface.go

```diff
@@ -0,0 +1,108 @@
+package integrationtests
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"os"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/msal"
+)
+
+type labClient struct {
+	labApplication *msal.ConfidentialClientApplication
+}
+
+/*
+Commenting this until it is used in a test case
+type app struct {
+	AppType      string `json:"appType"`
+	AppName      string `json:"appName"`
+	AppID        string `json:"appId"`
+	RedirectURI  string `json:"redirectUri"`
+	Authority    string `json:"authority"`
+	LabName      string `json:"labName"`
+	ClientSecret string `json:"clientSecret"`
+}
+*/
+
+type user struct {
+	AppID            string `json:"appId"`
+	ObjectID         string `json:"objectId"`
+	UserType         string `json:"userType"`
+	DisplayName      string `json:"displayName"`
+	Licenses         string `json:"licences"`
+	Upn              string `json:"upn"`
+	Mfa              string `json:"mfa"`
+	ProtectionPolicy string `json:"protectionPolicy"`
+	HomeDomain       string `json:"homeDomain"`
+	HomeUPN          string `json:"homeUPN"`
+	B2cProvider      string `json:"b2cProvider"`
+	LabName          string `json:"labName"`
+	LastUpdatedBy    string `json:"lastUpdatedBy"`
+	LastUpdatedDate  string `json:"lastUpdatedDate"`
+	Password         string
+}
+type secret struct {
+	Secret string `json:"value"`
+}
+
+func (l *labClient) getLabAccessToken() string {
+	if l.labApplication == nil {
+		secretS := os.Getenv("clientSecret")
+		secret, err := msal.CreateClientCredentialFromSecret(secretS)
+		if err != nil {
+			log.Fatal(err)
+		}
+		options := msal.DefaultConfidentialClientApplicationOptions()
+		clientID := os.Getenv("clientId")
+		options.Authority = MicrosoftAuthority
+		l.labApplication, err = msal.NewConfidentialClientApplication(clientID, secret, &options)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+	}
+	scopes := []string{"https://msidlab.com/.default"}
+	result, err := l.labApplication.AcquireTokenByClientCredential(context.Background(), scopes)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return result.GetAccessToken()
```

Comment:
This section gets called for every test, which means we make a network call each time to get a new token, since #48 has not been implemented. We should consider using acquire token silent or caching the token to avoid making that call every time.

Reply:
I have made the change to do an acquireTokenSilent before the acquireToken method, but looks like some stuff there is broken, so acquire token silent really does not return back a token even when there is one in the cache, I'll create an issue to track this.

Reply:
Tracked issue : https://github.com/AzureAD/microsoft-authentication-library-for-go/issues/106